src/app/src/main/java/com/example/bookmark/MyBookDetailsActivity.java

```diff
@@ -179,8 +191,43 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
             return;
         }
 
+        if (requestCode == GET_ISBN_TO_GIVE_BOOK || requestCode == GET_ISBN_TO_RECEIVE_BOOK) {
+            String isbn = data.getStringExtra("ISBN");
+            if (book.getIsbn().equals(isbn)) {
+                if (requestCode == GET_ISBN_TO_GIVE_BOOK) {
```

Comment:
This code here doesn't make sense if GET_ISBN_TO_GIVE_BOOK is equal to GET_ISBN_TO_RECEIVE_BOOK.